draft-ietf-acme-acme.md

```diff
@@ -1017,11 +1019,13 @@ encoded in the format specified in RFC 3339 {{!RFC3339}}.  This field is REQUIRE
 for objects with "valid" in the "status" field.
 
 challenges (required, array of objects):
-: For pending authorizations, the challenges that the client can fulfill in
-order to prove possession of the identifier.  For final authorizations (in the "valid" or "invalid" state), the
-challenges that were used.  Each array entry is an object with parameters
-required to validate the challenge.  A client should attempt to fulfill
-one of these challenges, and a server should consider any one of the challenges
+: For pending authorizations, the challenges that the client can
+fulfill in order to prove possession of the identifier.  For valid
+authorizations, the challenge that was validated.  For invalid
+authorizations, the challenges that were attempted and failed.  Each
```

Comment:
I think it should say "the challenge that was attempted and failed", singular "challenge" vs "challenges". The later text says: "A client should attempt to fulfill **one** of these challenges".

Reply:
I don't think there's anything to say that a server MUST fail the authz if a single challenge fails.

Reply:
That is, I think it's valid behavior for the server to keep the authz open and allow the client to try multiple challenges until it fulfills one or the server gives up.  In other other words, I read the "one" as attaching to the "fulfill", not the "attempt".

Reply:
We talked OOB about this and agree the specification is vague on this point.

I would be in favour of clarifying that an invalid challenge == an invalid authorization. This matches the Let's Encrypt implementation and when in doubt of the decision to make I'm firmly in favour of standardizing what has implementation experience in the client/server ecosystem today.